src/serializer/functions.js

```diff
@@ -295,19 +350,28 @@ export class Functions {
     return call.bind(this, globalThis, args);
   }
 
+  _optimizeFunction(funcValue: ECMAScriptSourceFunctionValue | BoundFunctionValue) {
+    let call = this._callOfFunction(funcValue);
+    let effects = this.realm.evaluatePure(() =>
+      this.realm.evaluateForEffectsInGlobalEnv(call, undefined, "additional function")
+    );
+    invariant(effects);
+    let additionalFunctionEffects = this._createAdditionalEffects(effects, true);
+    invariant(additionalFunctionEffects);
+    if (funcValue instanceof BoundFunctionValue) {
+      invariant(funcValue.$BoundTargetFunction instanceof ECMAScriptSourceFunctionValue);
+      this.writeEffects.set(funcValue.$BoundTargetFunction, additionalFunctionEffects);
```

Comment:
This seems wrong in general, as the effects only apply for whatever binding was in place here.

Reply:
I now see that `callOfFunction` does something similar. Then why not do it once at the top of `_optimizeFunction`?

Reply:
The issue is that the bound function's target function is the body I want to replace.